pdi-execution-engine/api/src/main/java/org/pentaho/di/engine/api/IEngine.java

```diff
@@ -0,0 +1,14 @@
+package org.pentaho.di.engine.api;
+
+/**
+ * An IEngine is responsible for executing an ITransformation.
+ *
+ * In order to do so, it needs to inspect the structure of that
+ * trans, rewrite or modify if necessary (leveraging IOperationVisitors),
+ * and "resolve" the trans IOperations to concrete functions (ICallableOperations).
+ */
+public interface IEngine {
+
+   void execute( ITransformation trans );
```

Comment:
Yep.  @ccaspanello is looking at parameters right now.

Context classes make me a bit nervous, since they can enable excessive class coupling.  But if we make sure to keep the ExecContext class(es) slim it will probably make things simpler.